doc/migration/version-upgrade-guide.md

```diff
@@ -168,8 +168,139 @@ The following sections outline the upgrade steps for each SDK version along with
     version = "=v0.7.1"
   ```
 
+## `v0.8.x`
+
+The SDK version `v0.8.x` supports scaffolding projects to use Go modules by default. It is recommended that you migrate your operator project to use modules for dependency management, however you can choose to keep using `dep`. The upgrade steps for both are outlined below:
+
+### `dep`
 
+- Update the SDK constraint in `Gopkg.toml` to version `v0.8.2`.
+  ```TOML
+  [[constraint]]
+    name = "github.com/operator-framework/operator-sdk"
+    version = "=v0.8.2"
+  ```
+- Pin the controller-tools dependency to the following revision. See the release notes or [#1278](https://github.com/operator-framework/operator-sdk/pull/1278/) for why this is needed.
+  ```TOML
+  [[override]]
+    name = "sigs.k8s.io/controller-tools"
+    revision = "9d55346c2bde73fb3326ac22eac2e5210a730207"
+  ```
+- Run `dep ensure` to update the vendor directory.
+
+### `modules`
+
+To get familiar with Go modules read the [modules wiki][modules-wiki]. In particular the section on [migrating to modules][migrating-to-modules].
+
+- Ensure that you have Go 1.12+ and [Mercurial][mercurial] 3.9+ installed.
```

Comment:
```suggestion
- Ensure that you have Go 1.12+ (I do not think that The Mercurial is required at all)
```

Reply:
Also, is required to:
-  create the `group.go`
- Ensure that ObjectStruct has

```
// +k8s:openapi-gen=true
// +kubebuilder:subresource:status// 
```

Reply:
Adding `group.go` isn't really necessary since it's not a breaking change. It only suppresses a warning from gengo when running the gen-code commands.

But you're right on the second point: I should highlight the need for `+kubebuilder` tags now that CRD manifests are overwritten in `v0.8.x` and will remove any manual edits made by users to the CRD.
From the CHANGELOG:
```
- Go operator CRDs are overwritten when being regenerated by operator-sdk generate openapi. 
Users can now rely on +kubebuilder annotations in their API code, which provide access to most OpenAPIv3
 validation properties (the full set will be supported in the near future, see this PR) and other CRD fields. (#1278)
```
This probably should have been marked as breaking change to begin with. Well spotted.

Reply:
Also I can't recall if I had to setup mercurial when I used Go modules but it's definitely part of our pre-reqs:
https://github.com/operator-framework/operator-sdk#prerequisites

Although it might be one of git or mercurial:
https://github.com/golang/go/wiki/GoGetTools

I'll need to double check and remove it from the SDK's pre-reqs if that's not the case.

Reply:
I'm fairly sure mercurial was/is required. I'm pretty sure some transitive dependency somewhere comes from a mercurial repo.

Also, just be careful about _which_ kubebuilder tags are mentioned about being needed/supported. We won't switch to controller-tools 0.2.x until the next release, so tags like `+kubebuilder:subresource:status` that were added in controller-tools 0.2.0 should not be mentioned.

Reply:
Yeah I've linked to the legacy kubebuilder docs to show that not all marker comments are supported for the controller-tools version tied SDK release v0.8.2.

But `// +kubebuilder:subresource:status` specifically is supported pre controller-tools v0.2.0 so I've kept that in the example.

Reply:
HI @joelanford, @hasbro17  I have no mercurial installed locally and all are working. 

```
$ mercurial
-bash: mercurial: command not found
```

Reply:
@camilamacedo86 The mercurial command is `hg`, so check if you have that.

Here's the original issue: https://github.com/operator-framework/operator-sdk/issues/1681

Looks like it affected 0.9 at least, not sure about 0.8 or 0.10.

Reply:
I lot of shame now :-P Yes, I have it! All explained tks for the patient with me.